components/missions/Missions.jsx

```diff
@@ -0,0 +1,229 @@
+            this.setState({
+              futureMissions
+            });
+          }
+      });
+    }
+
+    /* uploads mission and re-renders mission view.
+    Only allow adding NEW missions
+    */
+    uploadAndAddMission = (mission, url) => {
+      var newMissions = this.state.futureMissions;
+      db.collection("missions").add({
+        description: mission.description,
+        endDate: new firebase.firestore.Timestamp(mission.endDate, 0),
+        imageLink: url,
+        location: mission.location,
+        name: mission.name,
+        startDate: new firebase.firestore.Timestamp(mission.startDate, 0),
+        url: mission.url,
+      }).then((missionCallback) => {
+        // id needed for Firestore to update, imageLink retrieves image
+        mission.id = missionCallback.id;
+        mission.imageLink = url;
+        newMissions.push(mission);
+        this.setState({
+          futureMissions: newMissions
+        });
+      });
+    }
+
+    /* Callback to add Mission 
+      if photo already in FireStorage, use it else upload new photo.
+    */
+    addMission = (mission) => {
+      uploadImage(mission, this.uploadAndAddMission);
+    }
+
+    handleForm = (open) => {
+      this.setState({
+        openForm: open
+      });
+    }
+
+    render() {
+      const { classes } = this.props;
+      let addMissionForm; // only render form when done loading!
+      let data = [];
+
+      let loading = (<CircularProgress className={classes.progress} />);
+      // only display data if NOT loading
+      if (!this.state.loading) {
+          // show past
+          if (this.state.showPast == 1) {
+            data = this.state.pastMissions.map((mission) => (
+              (<Grid key={mission.id} item xs={12} md={4} lg={3}>
+                <Mission mission={mission} removeCallback={this.removeMission}/>
+            </Grid>)));
+          } else { // show present
+            data = this.state.futureMissions.map((mission) => (
+              (<Grid key={mission.id} item xs={12} md={4} lg={3}>
+                <Mission mission={mission} removeCallback={this.removeMission}/>
+            </Grid>)));
+          }
+          // needed to display empty Mission form
+          let emptyMission = {
+            description: '',
+            endDate: now.unix(),
+            imageLink: '',
+            location: '',
+            name: '',
+            startDate: now.unix(),
+            url: '',
+          };
+          addMissionForm = (<EditForm open={this.state.openForm}
+            updateMission={this.addMission} mission={emptyMission}
+            handleEdit={this.handleForm} update="false">
+          </EditForm>);
+      }
+
+      return (
+          <div>
+            <Button className={classes.newMissionButton} variant="contained" 
+            color="primary" onClick={() => this.handleForm(true)}>
+              New Mission
+            </Button>
+            {!this.state.loading && addMissionForm}
+            <Tabs

```

Comment:
When I ran the site locally the behavior for which tab to click to switch the state seemed to be backward. I had to click the currently active tab to switch to the other one. I'm not seeing an obvious cause in the code but this needs to be fixed before I approve the review.